src/shapes.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * @deprecated, will probably not use this
+ */
+class Shape {
+    constructor(rotation = 0) {
+        this.rotation = rotation; //rotation of shape in radians
+    }
+
+}
+
+export class Rect extends Shape {
+    constructor(height, width, rotation) {
+        super(rotation);
+        this.height = height;
+        this.width = width;
+    }
+
+    get clone() {
```

Comment:
I don't know how I feel about `clone` being a getter/property and not a regular method. 

```
const r = new Rect(10, 10, 0);
const clone = r.clone;

// This fails (and it should), but looking at the API
//    it isn't clear that r.clone creates a new clone every time you call it.
//    It looks like you're referencing the same clone each time. 
assert(clone === r.clone); 

```